### PR TITLE
News feed comment fix

### LINKED ISSUE #2190: Adding a comment to a news feed entry fails
**Describe the bug**
- Please work with @Ayush0Chaudhary for comments, advice and review of your PR.

**To Reproduce**

- https://youtu.be/86AzK_Y_GCQ?t=48

**Expected behavior**
- Comment created 

**Actual behavior**
- No comment created

**Screenshots**
- N/A

**Additional details**
- You must get the test **_code coverage up to 100%_** for any edited files. No functionality must be removed to achieve this goal.

### PR DESCRIPTION
**Pull request by imshivam-gupta**
**What kind of change does this PR introduce?**

Bugfix - Comments in newsfeed not visible before due to wrong api call.

**Issue Number:**

Fixes #2190 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

Graphql api used to fetch posts was altered.

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by palisadoes** [comment]
Please fix the failing test. 
- https://www.youtube.com/watch?v=dQuEKpsy1vM

**Comment by imshivam-gupta** [comment]
@palisadoes , I have not changed any of the files for which `codecov/patch` is failing. Can you please guide me why the test is failing?


image

**Comment by palisadoes** [comment]
Please merge with your upstream and make an updating commit.

**Comment by imshivam-gupta** [comment]
@palisadoes , I have successfully passed all the tests. Could you please assign reviewers for this pull request? I hope you find my work satisfactory.

**Review by noman2002 — COMMENTED** [review]
Please attach a video.

**Comment by imshivam-gupta** [comment]
@noman2002 You can see the output here - [Link](https://www.youtube.com/watch?v=deZpPhI46s8)

**Comment by imshivam-gupta** [comment]
@noman2002 Can you please review the request?

**Review by noman2002 — APPROVED** [review]
Looks good to me.

**Comment by imshivam-gupta** [comment]
@Kevoniat Can you also review the request?

**Comment by imshivam-gupta** [comment]
@palisadoes Can you review the request it is pending from last 5 days.

**Review by palisadoes — CHANGES_REQUESTED** [review]
@imshivam-gupta When you first make the comment it shows up at the top. Then you have to refresh to see the comment at the bottom of the list, where it should belong.

How can you ensure that the ordering is correct without the additional user action? This is not ideal.

**Comment by imshivam-gupta** [comment]
@palisadoes The change requested by you has been implemented in this latest commit. Here is the [Link](https://youtube.com/shorts/cSxEPi_fd3g?feature=share) to see output. 

@noman2002, Could you please re-review the request? The review became stale after the latest commit.

I appreciate your feedback on the code changes. If you have any further suggestions or corrections, feel free to let me know.

File: lib/view_model/widgets_view_models/comments_view_model.dart [hunk 1]
```diff
@@ -37,31 +50,47 @@ class CommentsViewModel extends BaseModel {
...
+  /// This function add comment locally.
+  ///
+  /// **params**:
+  /// * `msg`: BuildContext, contain parent info
+  ///
+  /// **returns**:
+  ///   None
   void addCommentLocally(String msg) {
     _postService.addCommentLocally(_postID);
     final creator = _userConfig.currentUser;
```
Comment (noman2002) [inline lib/view_model/widgets_view_models/comments_view_model.dart:96]: @imshivam-gupta Is this function sending the latest added comment to server or just adding it locally. If its adding the comment locally please send it to server.

Comment (imshivam-gupta) [reply lib/view_model/widgets_view_models/comments_view_model.dart:96]: it's sending to the server for permanent changes. Local addition is done for UI purpose to show the added comment at that moment.

File: lib/view_model/widgets_view_models/comments_view_model.dart [hunk 2]
```diff
@@ -37,31 +50,47 @@ class CommentsViewModel extends BaseModel {
...
   ///
-  /// params:
-  /// * `msg` : text of the comment to add.
-  Future createComment(String msg) async {
+  /// **returns**:
+  /// * `Future<void>`: promise that will be fulfilled when comment is added.
+  ///
+  Future<void> createComment(String msg) async {
     print("comment viewModel called");
     await _commentService.createComments(_postID, msg);
```
Comment (imshivam-gupta) [inline lib/view_model/widgets_view_models/comments_view_model.dart:83]: This line adds comment to the server

File: lib/view_model/widgets_view_models/comments_view_model.dart (hunk 1 above)
Comment (imshivam-gupta) [reply lib/view_model/widgets_view_models/comments_view_model.dart:96]: Another function was called using `_commentService` to add the latest comment to server also. I will comment the line in next comment.

**Review by Kevoniat — APPROVED** [review]

**Comment by palisadoes** [comment]
@imshivam-gupta Please include an updated video. Upload it into the comment box.

**Comment by imshivam-gupta** [comment]
[Link](https://youtube.com/shorts/cSxEPi_fd3g?feature=share) Latest Video

**Comment by palisadoes** [comment]
1. It still doesn't look like the comment is immediately updated at the end of the list without a manual refresh.
2. @noman2002 Please take a look

**Comment by imshivam-gupta** [comment]
In my device as soon as I post the comment it is visible on top and then if I visit the comment section again it is still showing that on top. @palisadoes can you please elaborate more in adding comment at the end of the list?

**Comment by palisadoes** [comment]
1. In most apps, the comments to a post are always visible in chronological order, the most recent at the end of the list of comments.
2. Your change makes the posts appear in reverse chronological order, with the most recent at the beginning. 

This makes it harder to understand the flow of a conversation as you have to read it from the bottom of the list of comments to the top, instead of scrolling from the top of the conversation downwards

**Comment by imshivam-gupta** [comment]
Ok Sir I got it . It was designed like that initially with local comments addition that's why I intentionally reversed the order in showing server comments also. I will fix it and  update this today and again push the latest commit

**Comment by imshivam-gupta** [comment]
@palisadoes - Can you please check this now - [Link](https://youtube.com/shorts/iJWx9pxqKP4?feature=share). I have changed the code so that latest comments adds on end of list.

**Comment by palisadoes** [comment]
@noman2002 @Kevoniat Please review again. The sort order looks OK now

**Review by noman2002 — APPROVED** [review]
Looks good to me.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**